packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts

```diff
@@ -0,0 +1,28 @@
+import { SandboxBackendIdResolver } from '../commands/sandbox/sandbox_id_resolver.js';
+import { BackendIdentifierResolver } from './backend_identifier_resolver.js';
+
+/**
+ * Resolves the backend id when branch or stack is passed as an arg, otherwise returns a sandbox backend identifier
+ */
+export class BackendIdentifierResolverWithFallback
```

Comment:
nit:
```suggestion
export class BackendIdentifierResolverChain
```
Could even generalize to taking in an array of resolvers and return the first one that doesn't return undefined. But that could also be a load of YAGNI

Reply:
I went down that road last night and decided it was YAGNI and walked it back